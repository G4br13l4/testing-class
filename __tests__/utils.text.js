const utils = require('../utils')

// describe('countWords', () => {
//   test('should', () => {
//     const result = utils.countWords({})
//     expect(result).toMatchSnapshot()
//   });

//   test('should', () => {
//     expect(() => {
//       utils.countWords()
//     }).not.toThrow();
//   });
// });

//llamadas a apis
describe('cafee', () => {
   test('should', async () => {
      const result = await utils.getCoffee()
      //jest.runAllTimers testea sin tener en cuenta los timers
      expect(result).toBe("emoji");
   });
});

// para testear ids, funciones con resuktado diferente

const getID = () => {
  return {
    id: expect.any (String),
    create_at: expect.any(Date)
    //anexar regex 
  }
}

// ver si una funcion si triggeo

jest.spyOn(utils, "getID")

describe('Name of the group', () =>{
    test('should', () => {
      const result = utils.getID()
      expect(utils.getID).toBeCalled()
    });

    );
});


o jest.fn()



