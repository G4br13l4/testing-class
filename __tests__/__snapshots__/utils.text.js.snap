// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`countWords should 1`] = `
Object {
  "la": 3,
}
`;
